client/src/connection/itc/CodeRunner.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright © 2024, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { commands } from "vscode";
+
+import { ITCSession } from ".";
+import { LogLine, getSession } from "..";
+import { useRunStore } from "../../store";
+
+class CodeRunner {
+  protected executionInterval: ReturnType<typeof setInterval>;
+
+  public async runCode(
+    code: string,
+    startTag: string = "",
+    endTag: string = "",
+  ): Promise<string> {
+    // If we're already executing code, lets wait for it
+    // to finish up.
+    if (useRunStore.getState().isExecutingCode) {
+      await new Promise((resolve) => {
+        if (this.executionInterval) {
+          clearInterval(this.executionInterval);
+        }
+
+        this.executionInterval = setInterval(() => {
```

Comment:
oh just noticed this, how about subscribe the state change event, like
https://github.com/sassoftware/vscode-sas-extension/blob/main/client/src/components/logViewer/index.ts#L91

Reply:
That works! Updated in https://github.com/sassoftware/vscode-sas-extension/pull/880/commits/bfce3044abf0021f961b85cd12cc21406209e729